specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,148 @@
+that can destroy and instantiate animations.
+2.	Create a new interface `IAnimatedVisualSource3` from which to get the new `IAnimatedvisual2`
+3.	Add a `CacheMode` property to AnimatedVisualPlayer that will control
+when animations should be destroyed and when they should be instantiated.
+
+
+# API Pages
+
+## AnimatedVisualPlayer.CacheMode property
+
+Gets or sets a value that determines how animations are cached. Defaults to 'Always'.
+
+The two modes for the property will specify the behavior the AnimatedVisualPlayer will have
+when the player is idle (when PlayAsync is not active).
+
+|Mode| Behavior|
+|---- | -------|
+|Always| AnimatedVisualPlayer will pre-create animations even before `PlayAsync()` is called,
+and not destroy any when player is idle. |
+|None | AnimatedVisualPlayer will not create animations until `PlayAsync()` is called,
+and will destroy them when it completes. |
+
+If you have an animation that will start on click, you may want to set `CacheMode` to 'Always'
+when the mouse enters this control, 
+and back to `None` when mouse leaves to ensure that the animation starts immediately on click,
+but don't use up resources otherwise.
+
+Setting the `CacheMode` to `Always` when the mouse enters the control will ensure that
+the animation objects are preloaded so that the animation may start immediately when it needs to play. 
+
+```c#
+private void Player_PointerEntered(object sender, PointerRoutedEventArgs e)
+{
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode.Always;
+}
+async private void Player_PointerExited(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode.None;
+}
+
+async private void Player_OnClick(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.PlayAsync(0.0, 1.0);
+}
+
+```
+
+If `PlayAsync` is called and the animations have not been instantiated due to `CacheMode` being set to `None`,
+`PlayAsync` will instantiate the animations so it can play the visual. 
+
+# API Details
+
+```c#
+namespace Microsoft.UI.Xaml.Controls
+{
+  unsealed runtimeclass AnimatedVisualPlayer
+  {
+      // Existing ...
+
+    //Sets the caching behavior of the AnimatedVisualPlayer
+    Microsoft.UI.Xaml.UIElement.CacheMode AnimationsCacheMode CacheMode;
+  }
+}
+```
+
+## IAnimatedVisual2 Interface
+
+An extension to
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+that adds the ability to control when animation objects are created and destroyed.
+
+| Method | Meaning |
+|-------------|---------|
+| InstantiateAnimations | Instantiates the animation visuals for the provided progress value of the animation. |
+| DestryAnimations | Destroys all animation visuals for a given animation.|
+
+```c# 
+interface IAnimatedVisual2 requires IAnimatedVisual
+{
+    void InstantiateAnimations(Double progressHint);
```

Comment:
Suggest `CreateAnimations`; the APIs today generally use 'create' rather than 'instantiate'.

Reply:
Agree, "CreateAnimations" is better. I was also thinking about "SuspendAnimations"+"RestoreAnimations" in case if there will be some API in future that will allow to suspend animations without actually destroying them

Reply:
I'm not attached to create/destroy, but "suspend" sounds like it's stopping or freezing the animation effect, rather than being about resources.

Reply:
Since this is really about caching, maybe PrepareAnimations / DiscardAnimations?

Reply:
@oldnewthing oh, I really like this naming

Reply:
Could also be Load/Unload: LoadAnimations/UnloadAnimations.

The naming is tricky because the class is "animated visual" so it seems weird that you tell the animated visual "unload your animations" -- isn't that just "unload everything"? But really, we want to unload just the animating bits, and leave the static bits of the animation.